### ISSUE TITLE
Does this work on the latest version of Julia? 

### ISSUE THREAD

**Issue by azev77** [matthieugomez/EconPDEs.jl#25]
I tried README example w/ Julia v 1.6.2: 
```julia
julia> using EconPDEs

julia> stategrid = (;μ = range(-0.05, 0.1, length = 500))
(μ = -0.05:0.0003006012024048096:0.1,)

julia> solend = (; V = ones(500))
(V = [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0  …  1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0],)

julia> function f(state::NamedTuple, sol::NamedTuple)
           μbar = 0.018 ; ϑ = 0.00073 ; θμ = 0.252 ; νμ = 0.528 ; ρ = 0.025 ; ψ = 1.5 ; γ = 7.5
           (; μ) = state
           (; V, Vμ_up, Vμ_down, Vμμ) = sol
           # note that pdesolve will directly compute the derivatives of the valuefunction.
           # up and down correspond to the upward and downard derivative obtained by first difference
           Vμ = (μ <= μbar) ? Vμ_up : Vμ_down
           Vt = - (1  - ρ * V + (1 - 1 / ψ) * (μ - 0.5 * γ * ϑ) * V + θμ * (μbar - μ) * Vμ +
           0.5 * νμ^2 * ϑ * Vμμ  + 0.5 * (1 / ψ - γ) / (1- 1 / ψ) * νμ^2 *  ϑ * Vμ^2/V)
           (; Vt)
       end
ERROR: syntax: invalid assignment location "; μ" around REPL[7]:3
Stacktrace:
 [1] top-level scope
   @ REPL[7]:1

julia> 
```

**Comment by matthieugomez** [comment]
Yeah sorry, it's Julia 1.7 —  I thought it was going to get released sooner. 
For now you can do
```Julia
function f(state::NamedTuple, sol::NamedTuple)
           μbar = 0.018 ; ϑ = 0.00073 ; θμ = 0.252 ; νμ = 0.528 ; ρ = 0.025 ; ψ = 1.5 ; γ = 7.5
           μ = state.μ
           V, Vμ_up, Vμ_down, Vμμ = sol.V, sol.Vμ_up, sol.Vμ_down, sol.Vμμ
           # note that pdesolve will directly compute the derivatives of the valuefunction.
           # up and down correspond to the upward and downard derivative obtained by first difference
           Vμ = (μ <= μbar) ? Vμ_up : Vμ_down
           Vt = - (1  - ρ * V + (1 - 1 / ψ) * (μ - 0.5 * γ * ϑ) * V + θμ * (μbar - μ) * Vμ +
           0.5 * νμ^2 * ϑ * Vμμ  + 0.5 * (1 / ψ - γ) / (1- 1 / ψ) * νμ^2 *  ϑ * Vμ^2/V)
           (Vt = Vt)
       end
```

**Comment by azev77** [comment]
Sorry, typo.
I meant I'm getting error w/ the latest release Julia v 1.6.2

**Comment by matthieugomez** [comment]
solved with https://github.com/matthieugomez/EconPDEs.jl/commit/9d7ecb7089a8ab5703def111e1b66c30d9d49f13

**Comment by azev77** [comment]
All kinds of other errors, hopefully will be resolved in v1.7, I can wait in the mean time: 
```julia
julia> ys, residual_norms = pdesolve(f, stategrid, solend, range(0, 1000, length = 100))
    Time Residual
-------- --------
ERROR: MethodError: no method matching _setindex!(::Matrix{ForwardDiff.Dual{ForwardDiff.Tag{EconPDEs.var"#G_helper!#4"{Vector{Bool}, EconPDEs.var"#23#35"{DataType, Matrix{Float64}, Matrix{Float64}, EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}}, Vector{Float64}, Float64}, Float64}, Float64, 3}}, ::Type{Type{(:V,)}}, ::ForwardDiff.Dual{ForwardDiff.Tag{EconPDEs.var"#G_helper!#4"{Vector{Bool}, EconPDEs.var"#23#35"{DataType, Matrix{Float64}, Matrix{Float64}, EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}}, Vector{Float64}, Float64}, Float64}, Float64, 3}, ::CartesianIndex{1})
Closest candidates are:
  _setindex!(::Any, ::Any, ::EconPDEs.StateGrid, ::Any, ::AbstractArray, ::AbstractArray) at C:\Users\azevelev\.julia\packages\EconPDEs\yRCOT\src\pdesolve.jl:136
  _setindex!(::AbstractArray, ::Type{Tsolution}, ::NamedTuple, ::CartesianIndex) where Tsolution at C:\Users\azevelev\.julia\packages\EconPDEs\yRCOT\src\pdesolve.jl:168
  _setindex!(::Any, ::AbstractArray) at C:\Users\azevelev\.julia\packages\EconPDEs\yRCOT\src\pdesolve.jl:130
Stacktrace:
  [1] hjb!(apm::typeof(f), stategrid::EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}, Tsolution::Type, ydot_M::Matrix{ForwardDiff.Dual{ForwardDiff.Tag{EconPDEs.var"#G_helper!#4"{Vector{Bool}, EconPDEs.var"#23#35"{DataType, Matrix{Float64}, Matrix{Float64}, EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}}, Vector{Float64}, Float64}, Float64}, Float64, 3}}, y_M::Matrix{ForwardDiff.Dual{ForwardDiff.Tag{EconPDEs.var"#G_helper!#4"{Vector{Bool}, EconPDEs.var"#23#35"{DataType, Matrix{Float64}, Matrix{Float64}, EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}}, Vector{Float64}, Float64}, Float64}, Float64, 3}}, bc_M::Matrix{Float64})
    @ EconPDEs C:\Users\azevelev\.julia\packages\EconPDEs\yRCOT\src\pdesolve.jl:160
  [2] hjb!(apm::Function, stategrid::EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}, Tsolution::Type, ydot::Vector{ForwardDiff.Dual{ForwardDiff.Tag{EconPDEs.var"#G_helper!#4"{Vector{Bool}, EconPDEs.var"#23#35"{DataType, Matrix{Float64}, Matrix{Float64}, EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}}, Vector{Float64}, Float64}, Float64}, Float64, 3}}, y::Vector{ForwardDiff.Dual{ForwardDiff.Tag{EconPDEs.var"#G_helper!#4"{Vector{Bool}, EconPDEs.var"#23#35"{DataType, Matrix{Float64}, Matrix{Float64}, EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}}, Vector{Float64}, Float64}, Float64}, Float64, 3}}, bc_M::Matrix{Float64}, ysize::Tuple{Int64, Int64})
    @ EconPDEs C:\Users\azevelev\.julia\packages\EconPDEs\yRCOT\src\pdesolve.jl:151
  [3] (::EconPDEs.var"#23#35"{DataType, Matrix{Float64}, Matrix{Float64}, EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}})(ydot::Vector{ForwardDiff.Dual{ForwardDiff.Tag{EconPDEs.var"#G_helper!#4"{Vector{Bool}, EconPDEs.var"#23#35"{DataType, Matrix{Float64}, Matrix{Float64}, EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}}, Vector{Float64}, Float64}, Float64}, Float64, 3}}, y::Vector{ForwardDiff.Dual{ForwardDiff.Tag{EconPDEs.var"#G_helper!#4"{Vector{Bool}, EconPDEs.var"#23#35"{DataType, Matrix{Float64}, Matrix{Float64}, EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}}, Vector{Float64}, Float64}, Float64}, Float64, 3}})
    @ EconPDEs C:\Users\azevelev\.julia\packages\EconPDEs\yRCOT\src\pdesolve.jl:56
  [4] (::EconPDEs.var"#G_helper!#4"{Vector{Bool}, EconPDEs.var"#23#35"{DataType, Matrix{Float64}, Matrix{Float64}, EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}}, Vector{Float64}, Float64})(ydot::Vector{ForwardDiff.Dual{ForwardDiff.Tag{EconPDEs.var"#G_helper!#4"{Vector{Bool}, EconPDEs.var"#23#35"{DataType, Matrix{Float64}, Matrix{Float64}, EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}}, Vector{Float64}, Float64}, Float64}, Float64, 3}}, y::Vector{ForwardDiff.Dual{ForwardDiff.Tag{EconPDEs.var"#G_helper!#4"{Vector{Bool}, EconPDEs.var"#23#35"{DataType, Matrix{Float64}, Matrix{Float64}, EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}}, Vector{Float64}, Float64}, Float64}, Float64, 3}})
    @ EconPDEs C:\Users\azevelev\.julia\packages\EconPDEs\yRCOT\src\finiteschemesolve.jl:9
  [5] forwarddiff_color_jacobian!(J::LinearAlgebra.Tridiagonal{Float64, Vector{Float64}}, f::EconPDEs.var"#G_helper!#4"{Vector{Bool}, EconPDEs.var"#23#35"{DataType, Matrix{Float64}, Matrix{Float64}, EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}}, Vector{Float64}, Float64}, x::Vector{Float64}, jac_cache::SparseDiffTools.ForwardColorJacCache{Vector{ForwardDiff.Dual{ForwardDiff.Tag{EconPDEs.var"#G_helper!#4"{Vector{Bool}, EconPDEs.var"#23#35"{DataType, Matrix{Float64}, Matrix{Float64}, EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}}, Vector{Float64}, Float64}, Float64}, Float64, 3}}, Vector{ForwardDiff.Dual{ForwardDiff.Tag{EconPDEs.var"#G_helper!#4"{Vector{Bool}, EconPDEs.var"#23#35"{DataType, Matrix{Float64}, Matrix{Float64}, EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}}, Vector{Float64}, Float64}, Float64}, Float64, 3}}, Vector{Float64}, Vector{Vector{Tuple{Float64, Float64, Float64}}}, Vector{Int64}, LinearAlgebra.Tridiagonal{Float64, Vector{Float64}}})
    @ SparseDiffTools C:\Users\azevelev\.julia\packages\SparseDiffTools\lC9Kz\src\differentiation\compute_jacobian_ad.jl:316
  [6] #2
    @ C:\Users\azevelev\.julia\packages\EconPDEs\yRCOT\src\finiteschemesolve.jl:16 [inlined]
  [7] (::NLSolversBase.var"#fj!#7"{EconPDEs.var"#G_helper!#4"{Vector{Bool}, EconPDEs.var"#23#35"{DataType, Matrix{Float64}, Matrix{Float64}, EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}}, Vector{Float64}, Float64}, EconPDEs.var"#2#5"{SparseDiffTools.ForwardColorJacCache{Vector{ForwardDiff.Dual{ForwardDiff.Tag{EconPDEs.var"#G_helper!#4"{Vector{Bool}, EconPDEs.var"#23#35"{DataType, Matrix{Float64}, Matrix{Float64}, EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}}, Vector{Float64}, Float64}, Float64}, Float64, 3}}, Vector{ForwardDiff.Dual{ForwardDiff.Tag{EconPDEs.var"#G_helper!#4"{Vector{Bool}, EconPDEs.var"#23#35"{DataType, Matrix{Float64}, Matrix{Float64}, EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}}, Vector{Float64}, Float64}, Float64}, Float64, 3}},
Vector{Float64}, Vector{Vector{Tuple{Float64, Float64, Float64}}}, Vector{Int64}, LinearAlgebra.Tridiagonal{Float64, Vector{Float64}}}, EconPDEs.var"#G_helper!#4"{Vector{Bool}, EconPDEs.var"#23#35"{DataType, Matrix{Float64}, Matrix{Float64}, EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}}, Vector{Float64}, Float64}}})(fx::Vector{Float64}, jx::LinearAlgebra.Tridiagonal{Float64, Vector{Float64}}, x::Vector{Float64})
    @ NLSolversBase C:\Users\azevelev\.julia\packages\NLSolversBase\GRQ1x\src\objective_types\abstract.jl:7
  [8] value_jacobian!!(obj::NLSolversBase.OnceDifferentiable{Vector{Float64}, LinearAlgebra.Tridiagonal{Float64, Vector{Float64}}, Vector{Float64}}, F::Vector{Float64}, J::LinearAlgebra.Tridiagonal{Float64, Vector{Float64}}, x::Vector{Float64})
    @ NLSolversBase C:\Users\azevelev\.julia\packages\NLSolversBase\GRQ1x\src\interface.jl:124
  [9] value_jacobian!!
    @ C:\Users\azevelev\.julia\packages\NLSolversBase\GRQ1x\src\interface.jl:122 [inlined]
 [10] newton_(df::NLSolversBase.OnceDifferentiable{Vector{Float64}, LinearAlgebra.Tridiagonal{Float64, Vector{Float64}}, Vector{Float64}}, initial_x::Vector{Float64}, xtol::Float64, ftol::Float64, iterations::Int64, store_trace::Bool, show_trace::Bool, extended_trace::Bool, linesearch::LineSearches.Static, linsolve::NLsolve.var"#29#31", cache::NLsolve.NewtonCache{Vector{Float64}})
    @ NLsolve C:\Users\azevelev\.julia\packages\NLsolve\gJL1I\src\solvers\newton.jl:48
 [11] #newton#9
    @ C:\Users\azevelev\.julia\packages\NLsolve\gJL1I\src\solvers\newton.jl:146 [inlined]
 [12] nlsolve(df::NLSolversBase.OnceDifferentiable{Vector{Float64}, LinearAlgebra.Tridiagonal{Float64, Vector{Float64}}, Vector{Float64}}, initial_x::Vector{Float64}; method::Symbol, xtol::Float64, ftol::Float64, iterations::Int64, store_trace::Bool, show_trace::Bool, extended_trace::Bool, linesearch::LineSearches.Static, linsolve::NLsolve.var"#29#31", factor::Float64, autoscale::Bool, m::Int64, beta::Int64, aa_start::Int64, droptol::Float64)
    @ NLsolve C:\Users\azevelev\.julia\packages\NLsolve\gJL1I\src\nlsolve\nlsolve.jl:23
 [13] implicit_timestep(G!::EconPDEs.var"#23#35"{DataType, Matrix{Float64}, Matrix{Float64}, EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}}, ypost::Vector{Float64}, Δ::Float64; is_algebraic::Vector{Bool}, iterations::Int64, verbose::Bool, method::Symbol, autodiff::Symbol, maxdist::Float64, J0c::Tuple{LinearAlgebra.Tridiagonal{Float64, Vector{Float64}}, Vector{Int64}}, y̲::Vector{Float64}, ȳ::Vector{Float64}, reformulation::S
ymbol)
    @ EconPDEs C:\Users\azevelev\.julia\packages\EconPDEs\yRCOT\src\finiteschemesolve.jl:24
 [14] pdesolve(apm::typeof(f), grid::Any, yend::Any, τs::StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}; is_algebraic::OrderedDict{Symbol, Bool}, bc::Nothing, verbose::Bool, kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ EconPDEs C:\Users\azevelev\.julia\packages\EconPDEs\yRCOT\src\pdesolve.jl:56
 [15] pdesolve(apm::Function, grid::Any, yend::Any, τs::StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}})
    @ EconPDEs C:\Users\azevelev\.julia\packages\EconPDEs\yRCOT\src\pdesolve.jl:17
 [16] top-level scope
    @ REPL[8]:1

julia> y, residual_norm =  pdesolve(f, stategrid, solend)
ERROR: MethodError: no method matching _setindex!(::Matrix{Float64}, ::Type{Type{(:V,)}}, ::Float64, ::CartesianIndex{1})
Closest candidates are:
  _setindex!(::Any, ::Any, ::EconPDEs.StateGrid, ::Any, ::AbstractArray, ::AbstractArray) at C:\Users\azevelev\.julia\packages\EconPDEs\yRCOT\src\pdesolve.jl:136
  _setindex!(::AbstractArray, ::Type{Tsolution}, ::NamedTuple, ::CartesianIndex) where Tsolution at C:\Users\azevelev\.julia\packages\EconPDEs\yRCOT\src\pdesolve.jl:168
  _setindex!(::Any, ::AbstractArray) at C:\Users\azevelev\.julia\packages\EconPDEs\yRCOT\src\pdesolve.jl:130
Stacktrace:
 [1] hjb!(apm::typeof(f), stategrid::EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}, Tsolution::Type, ydot_M::Matrix{Float64}, y_M::Matrix{Float64}, bc_M::Matrix{Float64})
   @ EconPDEs C:\Users\azevelev\.julia\packages\EconPDEs\yRCOT\src\pdesolve.jl:160
 [2] hjb!(apm::Function, stategrid::EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}, Tsolution::Type, ydot::Vector{Float64}, y::Vector{Float64}, bc_M::Matrix{Float64}, ysize::Tuple{Int64, Int64})
   @ EconPDEs C:\Users\azevelev\.julia\packages\EconPDEs\yRCOT\src\pdesolve.jl:151
 [3] #24
   @ C:\Users\azevelev\.julia\packages\EconPDEs\yRCOT\src\pdesolve.jl:66 [inlined]
 [4] finiteschemesolve(G!::EconPDEs.var"#24#36"{typeof(f), DataType, Matrix{Float64}, Matrix{Float64}, EconPDEs.StateGrid{Float64, 1, NamedTuple{(:μ,), Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}}}}, y0::Vector{Float64}; Δ::Float64, is_algebraic::Vector{Bool}, iterations::Int64, inner_iterations::Int64, verbose::Bool, inner_verbose::Bool, method::Symbol, autodiff::Symbol, maxdist::Float64, scale::Float64, J0c::Tuple{LinearAlgebra.Tridiagonal{Float64, Vector{Float64}}, Vector{Int64}}, minΔ::Float64, y̲::Vector{Float64}, ȳ::Vector{Float64}, reformulation::Symbol)
   @ EconPDEs C:\Users\azevelev\.julia\packages\EconPDEs\yRCOT\src\finiteschemesolve.jl:34
 [5] pdesolve(apm::typeof(f), grid::Any, yend::Any, τs::Nothing; is_algebraic::OrderedDict{Symbol, Bool}, bc::Nothing, verbose::Bool, kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ EconPDEs C:\Users\azevelev\.julia\packages\EconPDEs\yRCOT\src\pdesolve.jl:66
 [6] pdesolve(apm::Function, grid::Any, yend::Any, τs::Nothing) (repeats 2 times)
   @ EconPDEs C:\Users\azevelev\.julia\packages\EconPDEs\yRCOT\src\pdesolve.jl:17
 [7] top-level scope
   @ REPL[9]:1

julia> 
```

**Comment by matthieugomez** [comment]
look at new readme
